<a href="https://colab.research.google.com/github/jaimeload/HW_Terver/blob/main/hw7.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
from scipy import stats

Выбрать тест и проверить, есть  ли различия между выборками:

##1
Даны две  независимые выборки. Не соблюдается условие нормальности

x1  380, 420, 290

y1  140, 360, 200, 900

In [4]:
x1 = np.array([380, 420, 290])
y1 = np.array([140, 360, 200, 900])
α = 0.05

In [3]:
stats.mannwhitneyu(x1, y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

    0.63 > 0.05
    p-value > α ==> Cтатистически значимых различий на уровне значимости α нет

##2
Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

1е измерение до приема препарата: 150, 160, 165, 145, 155

2е измерение через 10 минут: 140, 155, 150,  130, 135

3е измерение через 30 минут: 130, 130, 120, 130, 125

In [5]:
before = np.array([150, 160, 165, 145, 155])
after10 = np.array([140, 155, 150, 130, 135])
after30 = np.array([130, 130, 120, 130, 125])
α = 0.05

In [6]:
stats.friedmanchisquare(before, after10, after30)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

    0.008 < 0.05
    p-value < α ==> Cтатистически значимые различия на уровне значимости α есть

##3
Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

In [7]:
stats.wilcoxon(before, after10)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

    0.0625 > 0.05
    p-value > α ==> Cтатистически значимых различий на уровне значимости α нет

##4
Даны 3 группы  учеников плавания.

В 1 группе время на дистанцию 50 м составляют:
56, 60, 62, 55, 71, 67, 59, 58, 64, 67

Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53

Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

In [10]:
g1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
g2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
g3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])
α = 0.05

In [11]:
stats.kruskal(g1, g2, g3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

    0.065 > 0.05
    p-value > α ==> Cтатистически значимых различий на уровне значимости α нет

##5
Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

    H₀: μ = 2.5
    H₁: μ != 2.5

In [17]:
balls = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
X = np.mean(balls)
μ = 2.5
α = 0.05
σ = np.std(balls, ddof=1)
n = len(balls)
X, σ, n

(2.5279999999999996, 0.1572542173961923, 10)

In [18]:
tn = (X - μ) / (σ / np.sqrt(n))
tn

0.5630613661802959

In [21]:
tk1 = stats.t.ppf(α / 2, n - 1)
tk2 = stats.t.ppf(1 - α / 2, n - 1)
tk1, tk2

(-2.262157162740992, 2.2621571627409915)

    tn не попадает в критическую область tk ==> H₀ не отвергается, а значит на уровне значимости α = 0.05
    недостаточно доказательств, чтобы утверждать, что среднее арифметическое отличается от 2.5